src/jsifier.js

```diff
@@ -241,7 +241,7 @@ function JSify(data, functionsOnly) {
         EXPORTED_FUNCTIONS[finalName] = 1;
         Functions.libraryFunctions[finalName] = 2;
       }
-      if ((EXPORT_ALL || (finalName in EXPORTED_FUNCTIONS)) && !noExport) {
+      if (EXPORT_ALL && !(finalName in EXPORTED_FUNCTIONS) && !noExport) {
```

Comment:
Interesting, so this line is only needed for EXPORT_ALL? But normal exporting is done elsewhere? Looks like not in this file, so I am guessing in emscripten.py.

This looks like a good change to me, I'm running some local tests and will merge assuming all is good. We can probably do even better though by moving the EXPORT_ALL logic into emscripten.py, that might be worth investigating.